lib/puppet/provider/uri_parsing.rb

```diff
@@ -16,8 +16,8 @@
 
 require File.expand_path(File.join(File.dirname(__FILE__), '../provider', 'login'))
 
-def uri_validation(data)
-  @client = OneviewSDK::Client.new(login)
+def uri_validation(data, client)
+  @client = client.nil? ? OneviewSDK::Client.new(login) : client
```

Comment:
I would like to use the previously created client, which can be either OneviewSDK::Client or OneviewSDK::ImageStreamer::Client

For a non-breaking change, I'm creating a new one when it's not provided.